pkg/operator/operator.go

```diff
@@ -123,7 +123,15 @@ func NewOperator() (context.Context, *Operator) {
 
 	// Client Config
 	config := ctrl.GetConfigOrDie()
-	config.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(options.FromContext(ctx).KubeClientQPS), options.FromContext(ctx).KubeClientBurst)
+	// Copy the leader config for lower QPS/Burst
+	// We changed this from explicitly setting the RateLimiter on the config and not creating
+	// a separate leaderConfig ourselves because this caused a subtle bug when copying the leaderConfig
+	// for the leader election client. The leaderConfig would use the same RateLimiter, so client-side rate
+	// limiting on the regular config would also cause client-side rate limiting on the leader election client,
+	// often leading to leader loss during large scale-ups or periods of high churn
+	leaderConfig := rest.CopyConfig(config)
```

Comment:
Should we still use the same QPS and burst for the leader election config? Doesn't seem like it should be necessary, but I'd have to look through the controller-runtime code to understand if there are any cases where we may hammer calls for the lease.

Reply:
I'd trust the default -- to me, it doesn't make sense that we would need higher QPS for the lease renewal -- it's a call that happens every 2s with backoff retry